src/Features/Core/Portable/ExtractMethod/MethodExtractor.VariableSymbol.cs

```diff
@@ -75,6 +72,23 @@ protected VariableSymbol(Compilation compilation, ITypeSymbol type)
 
                 return left.DisplayOrder - right.DisplayOrder;
             }
+
+            private static bool IsCancellationToken(ITypeSymbol originalType)
+            {
+                return originalType is
+                {
+                    Name: nameof(CancellationToken),
+                    ContainingNamespace:
+                    {
+                        Name: nameof(System.Threading),
+                        ContainingNamespace:
+                        {
+                            Name: nameof(System),
+                            ContainingNamespace.IsGlobalNamespace: true,
+                        }
+                    }
+                };
+            }
```

Comment:
simplified this, so we don't need to go do complex checks that require passing compilations around.